homeassistant/util/yaml.py

```diff
@@ -162,18 +160,22 @@ def _secret_yaml(loader: SafeLineLoader,
                                   " but 'logger: %s' found", logger)
                 del secrets['logger']
         else:
-            loader._SECRET_CACHE[secret_path] = None
-    secrets = loader._SECRET_CACHE[secret_path]
+            _SECRET_CACHE[secret_path] = {}
+    secrets = _SECRET_CACHE[secret_path]
 
     # Retrieve secret, first from secrets.yaml, then from keyring
     if secrets is not None and node.value in secrets:
         _LOGGER.debug('Secret %s retrieved from secrets.yaml.', node.value)
         return secrets[node.value]
-    for sname, sdict in loader._SECRET_CACHE.items():
-        if node.value in sdict:
+
+    while os.path.exists(secret_path) and secret_path in _SECRET_CACHE:
```

Comment:
But only if I originally meant what you think I'm saying...

probably test if the secrets file exists and then even load it, but we must probably be careful not to go above the config directory?